qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -384,3 +390,207 @@ def run(self, run_input: List[QuantumCircuit], **options) -> FakeJob:
+        circ_idx: int,
+    ) -> Dict[str, Any]:
+        """
+        Generate data for the circuit.
+        Args:
+            list_exp_dict (List): A List of dictionaries, each dictionary contains data of an experiment.
+            circ_idx (int): The circuit number we simulate.
+
+        Returns:
+            A dictionary that's filled with the simulated data.
+
+        Raises:
+            QiskitError: Raising an error if in the experiment running option, classified data is
+            requested.
+        """
+        circ_qubit_list = []
+        for exp_dict in list_exp_dict:
+            if circ_idx < exp_dict["num_circuits"]:
+                circ_qubit_list = circ_qubit_list + list(exp_dict["physical_qubits"])
+
+        shots = self.options.get("shots")
+        meas_level = self.options.get("meas_level")
+        meas_return = self.options.get("meas_return")
+        run_result = {}
+
+        if meas_level == MeasLevel.KERNELED:
+            memory = self._parallel_draw_iq_shots(list_exp_dict, shots, circ_qubit_list, circ_idx)
+            if meas_return == "avg":
+                memory = np.average(np.array(memory), axis=0).tolist()
+
+            run_result["memory"] = memory
+        else:
+            # The backend doesn't currently support 'meas_level = MeasLevel.CLASSIFIED'.
+            raise QiskitError("Classified data generator isn't supported for this backend")
+
+        return run_result
+
+    def run(self, run_input: List[QuantumCircuit], **run_options) -> FakeJob:
+        """
+        Run the IQ backend.
+        Args:
+            run_input: A list of QuantumCircuit for which the backend will generate
+             data.
+            **run_options: Experiment running options. The options that are supported in this backend are
+             'meas_level', 'meas_return' and 'shots'.
+                'meas_level': To generate data in the IQ plane, 'meas_level' should be assigned 1 or
+                    MeasLevel.KERNELED. The backend currently doesn't support 'meas_level' = 2  or
+                    MeasLevel.CLASSIFIED.
+                'meas_return': This option will only take effect if 'meas_level' = MeasLevel.CLASSIFIED.
+                    It can get either MeasReturnType.AVERAGE or MeasReturnType.SINGLE. For the value
+                      MeasReturnType.SINGLE the data of each shot will be stored in the result. For
+                      MeasReturnType.AVERAGE, an average of all the shots will be calculated and stored
+                      in the result.
+                'shots': The number of times the circuit will run.
+
+        Returns:
+            FakeJob: A job that contains the simulated data.
+
+        Raises:
+            QiskitError: Raised if the user try to run the experiment without setting a helper.
+        """
+
+        if not self.experiment_helper:
+            raise QiskitError("The backend `experiment_helper` attribute cannot be 'None'.")
```

Comment:
Check if there are additional methods where this check should be added

Reply:
Added also to `MockIQBackend`.